packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -509,6 +515,30 @@ export function scheduleUpdateOnFiber(
   // Mark that the root has a pending update.
   markRootUpdated(root, lane, eventTime);
 
+  if (enableProfilerNestedUpdateScheduledHook) {
+    if (
+      executionContext === CommitContext &&
+      root === rootCommittingMutationOrLayoutEffects
```

Comment:
This check is a little redundant, since `rootCommittingMutationOrLayoutEffects` is only set during commit.

Reply:
Should we warn if it's not equal then?

Reply:
Doesn't seem necessary. I was just mentioning that my conditional is a little redundant.